heron/healthmgr/src/java/com/twitter/heron/healthmgr/policy/AutoRestartBackpressureContainerPolicy.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.twitter.heron.healthmgr.policy;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import com.microsoft.dhalion.api.IResolver;
+import com.microsoft.dhalion.diagnoser.Diagnosis;
+import com.microsoft.dhalion.events.EventHandler;
+import com.microsoft.dhalion.events.EventManager;
+import com.microsoft.dhalion.policy.HealthPolicyImpl;
+
+import com.twitter.heron.healthmgr.HealthPolicyConfig;
+import com.twitter.heron.healthmgr.common.HealthManagerEvents.ContainerRestart;
+import com.twitter.heron.healthmgr.detectors.BackPressureDetector;
+import com.twitter.heron.healthmgr.detectors.LargeWaitQueueDetector;
+import com.twitter.heron.healthmgr.detectors.ProcessingRateSkewDetector;
+import com.twitter.heron.healthmgr.detectors.WaitQueueDisparityDetector;
+import com.twitter.heron.healthmgr.diagnosers.DataSkewDiagnoser;
+import com.twitter.heron.healthmgr.diagnosers.SlowInstanceDiagnoser;
+import com.twitter.heron.healthmgr.diagnosers.UnderProvisioningDiagnoser;
+import com.twitter.heron.healthmgr.resolvers.RestartContainerResolver;
+
+import static com.twitter.heron.healthmgr.HealthPolicyConfigReader.PolicyConfigKey.HEALTH_POLICY_INTERVAL;
+import static com.twitter.heron.healthmgr.diagnosers.BaseDiagnoser.DiagnosisName.DIAGNOSIS_DATA_SKEW;
+import static com.twitter.heron.healthmgr.diagnosers.BaseDiagnoser.DiagnosisName.DIAGNOSIS_SLOW_INSTANCE;
+import static com.twitter.heron.healthmgr.diagnosers.BaseDiagnoser.DiagnosisName.DIAGNOSIS_UNDER_PROVISIONING;
+
+public class AutoRestartBackpressureContainerPolicy extends HealthPolicyImpl
+    implements EventHandler<ContainerRestart> {
+
+  private static final String CONF_WAIT_INTERVAL_MILLIS =
```

Comment:
The variable name has `millis` as the unit whereas the config name as `min` as the unit. In absence of conversion this will be incorrect?

Reply:
changed `min` to `ms`
there is the same issue in `DynamicResourceAllocationPolicy`